client/signup/steps/domains/index.jsx

```diff
@@ -255,7 +239,7 @@ class DomainsStep extends React.Component {
 	};
 
 	submitWithDomain = ( googleAppsCartItem, shouldHideFreePlan = false ) => {
-		const shouldHideFreePlanItem = this.isEligibleVariantForDomainTest()
+		const shouldHideFreePlanItem = this.getIsPlanSelectionUnavailableInFlow()
 			? { shouldHideFreePlan }
 			: {};
```

Comment:
This code looks suspicious and could be another source of mysterious issues.

The `domains` and `domains-launch` steps that use this component declare that they provide the `shouldHideFreePlan` dependency. Therefore they should always provide it, even if the value is `false`, when calling `submitStep`.

But here, the dependency is submitted only when `getIsPlanSelectionUnavailableInFlow()` is `true`. That could lead to another `_assertFlowHasValidDependencies` fail, because the `domains-launch` step didn't fulfill its contract.

Additionally, the `getIsPlanSelectionUnavailableInFlow()` seems to be reverted: it submits the `shouldHideFreePlan` dependency only if there is no `plans*` step in the flow. But the `plans*` steps are the only ones that actually consume the `shouldHideFreePlan` dependency. And they won't get it.

Reply:
From what I gathered this means that we can remove this check completely from the  code and we actually should always pass this object either way. @niranjan-uma-shankar Do you agree?

Reply:
And you care correct @jsnajdr the condition should be the other way around in any case.